src/util/templates/ExtensionBasedTemplateEngine.ts

```diff
@@ -0,0 +1,28 @@
+import { NotImplementedHttpError } from '../errors/NotImplementedHttpError';
+import { getExtension } from '../PathUtil';
+import type { TemplateEngineInput } from './TemplateEngine';
+import { TemplateEngine } from './TemplateEngine';
+import { getTemplateFilePath } from './TemplateUtil';
+import Dict = NodeJS.Dict;
+
+export abstract class ExtensionBasedTemplateEngine<T extends Dict<any> = Dict<any>> extends TemplateEngine<T> {
+  protected readonly supportedExtensions: string[];
+
+  protected constructor(supportedExtensions: string[]) {
+    super();
+    this.supportedExtensions = supportedExtensions;
+  }
+
+  public async canHandle({ template }: TemplateEngineInput<T>): Promise<void> {
+    if (typeof template === 'undefined') {
+      throw new NotImplementedHttpError('No template was provided.');
+    }
+    // If the engine specifies supported extensions, check if the target template extension is included.
+    if (typeof this.supportedExtensions !== 'undefined') {
```

Comment:
This should be impossible since it is not an optional parameter. And if it was supposed to be an optional parameter I would go for `Array.isArray(this.supportedExtensions)`.